src/Workspaces/Core/Portable/Workspace/Host/Mef/MefLanguageServices.cs

```diff
@@ -65,7 +65,8 @@ internal bool TryGetService(Type serviceType, out Lazy<ILanguageService, Languag
             {
                 service = ImmutableInterlocked.GetOrAdd(ref _serviceMap, serviceType, svctype =>
                 {
-                    return PickLanguageService(_services.Where(lz => lz.Metadata.ServiceType == svctype.AssemblyQualifiedName));
```

Comment:
Should things like this get a ```// Perf``` comment so we know not to inline it again? It'd be nice if we could tell when a property incurs allocations on every access and then have analyzers to pull them out of loops/linq expressions like this, but without that it's hard to prevent someone from pushing this back into the Where.